cloud_governance/policy/azure/cleanup/unused_nat_gateway.py

```diff
@@ -13,6 +12,26 @@ def __init__(self):
         super().__init__()
         self.__active_cluster_ids = self._get_active_cluster_ids()
 
+    def __check_nat_gateway_metrics(self, resource_id: str):
+        """
+        This method returns bool by verifying nat metrics
+        :param resource_id:
+        :type resource_id:
+        :return:
+        :rtype:
+        """
+        metrics_data = self.monitor_operations.get_resource_metrics(resource_id=resource_id,
+                                                                    metricnames='SNATConnectionCount',
+                                                                    aggregation='Average')
+        if metrics_data.get('value'):
+            metrics_time_series_data = metrics_data.get('value', [])[0].get('timeseries', [])
+            if metrics_time_series_data:
+                for metric_time_frame in metrics_time_series_data:
+                    for data in metric_time_frame.get('data'):
+                        if data.get('average', 0) > 0:
```

Comment:
Are we expect to 0 for unused nat gateway ?

Reply:
Returning false, if it I found any active connection.